spec/rake_tasks/dummy_rake_spec.rb

```diff
@@ -5,30 +5,28 @@
     let(:task_name) { "dummy:do_something_once" }
     let(:task) { Rake::Task[task_name] }
 
-    before do
-      # clear the rake task from the memory to ensure it's not loaded multiple times
-      Rake::Task[task_name].clear if Rake::Task.task_defined?(task_name)
+    context 'when Rake.application.rake_require is used to load rake task' do
+      before do
+        Rake.application.rake_require("tasks/dummy")
+        Rake::Task.define_task(:environment)
+      end
 
-      # loaad the rake task only once
-      Rake.load_rakefile("tasks/dummy.rake")
-      Rake::Task.define_task(:environment)
-    end
-
-    after do
-      Rake::Task[task_name].reenable
+      after do
+        Rake::Task[task_name].reenable
 
-      # reset the state that was changed by the rake task execution
-      DummyOutput.count = 0
-    end
+        # reset the state that was changed by the rake task execution
+        DummyOutput.count = 0
```

Comment:
recommendation

As above

```suggestion
        DummyOutput.count = 0
```